modules/photo/include/opencv2/photo/ccm.hpp

```diff
@@ -0,0 +1,295 @@
+                        /// \f$
+                        /// M_{CCM}=
+                        /// \begin{bmatrix}
+                        /// k_R & 0 & 0\\
+                        /// 0 & k_G & 0\\
+                        /// 0 & 0 & k_B\\
+                        /// \end{bmatrix}
+                        /// \f$\n
+                        /// where\n
+                        /// \f$
+                        /// k_R=mean(R_{li}')/mean(R_{li})\\
+                        /// k_R=mean(G_{li}')/mean(G_{li})\\
+                        /// k_R=mean(B_{li}')/mean(B_{li})
+                        /// \f$
+    INITIAL_METHOD_LEAST_SQUARE,       ///<the least square method is an optimal solution under the linear RGB distance function
+};
+/** @brief  Macbeth and Vinyl ColorChecker with 2deg D50
+*/
+enum ColorCheckerType {
+    COLORCHECKER_Macbeth,                ///< Macbeth ColorChecker
+    COLORCHECKER_Vinyl,                  ///< DKK ColorChecker
+    COLORCHECKER_DigitalSG,              ///< DigitalSG ColorChecker with 140 squares
+};
+enum COLOR_SPACE {
+    COLOR_SPACE_sRGB,                       ///< https://en.wikipedia.org/wiki/SRGB , RGB color space
+    COLOR_SPACE_sRGBL,                      ///< https://en.wikipedia.org/wiki/SRGB , linear RGB color space
+    COLOR_SPACE_AdobeRGB,                   ///< https://en.wikipedia.org/wiki/Adobe_RGB_color_space , RGB color space
+    COLOR_SPACE_AdobeRGBL,                  ///< https://en.wikipedia.org/wiki/Adobe_RGB_color_space , linear RGB color space
+    COLOR_SPACE_WideGamutRGB,               ///< https://en.wikipedia.org/wiki/Wide-gamut_RGB_color_space , RGB color space
+    COLOR_SPACE_WideGamutRGBL,              ///< https://en.wikipedia.org/wiki/Wide-gamut_RGB_color_space , linear RGB color space
+    COLOR_SPACE_ProPhotoRGB,                ///< https://en.wikipedia.org/wiki/ProPhoto_RGB_color_space , RGB color space
+    COLOR_SPACE_ProPhotoRGBL,               ///< https://en.wikipedia.org/wiki/ProPhoto_RGB_color_space , linear RGB color space
+    COLOR_SPACE_DCI_P3_RGB,                 ///< https://en.wikipedia.org/wiki/DCI-P3 , RGB color space
+    COLOR_SPACE_DCI_P3_RGBL,                ///< https://en.wikipedia.org/wiki/DCI-P3 , linear RGB color space
+    COLOR_SPACE_AppleRGB,                   ///< https://en.wikipedia.org/wiki/RGB_color_space , RGB color space
+    COLOR_SPACE_AppleRGBL,                  ///< https://en.wikipedia.org/wiki/RGB_color_space , linear RGB color space
+    COLOR_SPACE_REC_709_RGB,                ///< https://en.wikipedia.org/wiki/Rec._709 , RGB color space
+    COLOR_SPACE_REC_709_RGBL,               ///< https://en.wikipedia.org/wiki/Rec._709 , linear RGB color space
+    COLOR_SPACE_REC_2020_RGB,               ///< https://en.wikipedia.org/wiki/Rec._2020 , RGB color space
+    COLOR_SPACE_REC_2020_RGBL,              ///< https://en.wikipedia.org/wiki/Rec._2020 , linear RGB color space
```

Comment:
The same note about capital letters,
e.g. `COLOR_SPACE_AppleRGB` -> `COLOR_SPACE_APPLE_RGB`